README.tmpl.md

```diff
@@ -0,0 +1,206 @@
+if you are constructing the file by hand).
+
+`gospel` will read _all_ `.words` files found at module roots in for the
+packages that are being checked and build a dictionary from them to give to
+hunspell for the spelling checks.
+
+Hunspell dictionaries are able to express more than just word matches though,
+and are able to indicate some grammatically related sets of words based on
+rules. This is covered lightly [below](#hunspell-dictionaries).
+
+
+### `.gospel.conf`
+
+Runtime behaviour of `gospel` can be modified in a persistent way through the
+TOML format `.gospel.conf` file. A number of options are provide:
+
+- `ignore_idents` — whether to include syntax information from the source code in the dictionary of acceptable words.
+- `lang` — the language tag to specify language locale.
+- `show` — whether to show context for identified misspellings.
+- `check_strings` — whether to check string literals.
+- `check_embedded` — whether to check spelling in files embedded using `//go:embed`.
+- `ignore_upper` — whether to ignore words that are all uppercase or their plurals.
+- `ignore_single` — whether to ignore single rune words.
+- `ignore_numbers` — whether to ignore number literals.
+- `mask_flags` — whether words that could be command-line flags should be removed prior to checking.
+- `mask_urls` — whether URLs should be removed prior to checking.
+- `check_urls` — whether the HTTP/HTTPS reachability of URLs should be checked.
+- `camel` — whether to split camelCase words into the components if the complete word is not accepted, otherwise split only on underscore.
+- `max_word_len` — the maximum length of words that should be checked.
+- `min_naked_hex` — minimum length for exclusion of words that are composed of only hex digits 0-9 and a-f (case insensitive).
+- `suggest` — when suggestions should be presented for misspellings: 0 - never, 1 - only the first time the error is found, 2 - always.
+- `diff_context` — how many lines around a change should be checked when the `-since` flag is used.
+- `entropy_filter` — controls the entropy filter used to exclude non-natural language from checking.
+    - `min_len_filtered` — the minimum length of text chunks to be considered by the entropy filter; the string literal length for strings, the file length for embedded files and the line or block length for comments.
+    - `entropy_filter.accept` — the range of complexity to allow as natural language for checking and roughly corresponds to the effective alphabet size for the language.
+
+The `.gospel.conf` file is intended to set base behaviour that can be
+modified with the command line flags during tuning.
+
+The current default `.gospel.conf` file looks like this:
+
+```toml
+{{.config -}}
+```
+
+## Hunspell Dictionaries
+
+Hunspell dictionaries are composed of two parts, a word list and an affix
+definition file. These are briefly described below. More information can be
+found in `man (5) hunspell`.
```

Comment:
nit: I would just type `man 5 hunspell`. the argument 5 isn't optional at least for me, because `man hunspell` will default to `man 1 hunspell`. and if someone doesn't get the syntax hint, they might actually type `man (5) hunspell` into their shell, which is invalid.

Reply:
I'm old. I read the as man, section 5, hunspell, not a command. I will change it.

Reply:
I didn't know about `man (N) NAME` being a format people use :) the fact that it's backquoted made me think you intended this as a shell command

Reply:
Indeed, there was inconsistency there.